web/src/stream_data.ts

```diff
@@ -471,6 +471,53 @@ export function is_new_stream_announcements_stream_muted(): boolean {
     return is_muted(realm.realm_new_stream_announcements_stream_id);
 }
 
+export function has_metadata_access(sub: StreamSubscription): boolean {
```

Comment:
I think this is a fine function to have. But it might want to have a comment noting that we expect this to be true in practice for every -- the server should be preventing a `StreamSubscription` from reaching clients without metadata access.